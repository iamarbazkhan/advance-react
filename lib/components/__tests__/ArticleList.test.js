import React from "react";
import ArticleList from "../ArticleList";
import renderer from 'react-test-renderer'; // it is used to create react component snapshots

const testProps={
    articles:{
        a:{
            id:'a'
        },
        b:{id:'b'}

    },
    articleActions:{
        lookupAuthor: jest.fn(()=>({}))
    }
}

describe("ArticleList",()=>{
    it('renders correctly',()=>{
       const tree= renderer.create(
            <ArticleList
            {...testProps}
            />
        ).toJSON()
        expect(tree).toMatchSnapshot();
    })
})