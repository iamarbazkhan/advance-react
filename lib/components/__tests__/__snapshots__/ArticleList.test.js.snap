// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div
    style={
      {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "marginBottom": 1,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        {
          "color": "#888",
          "fontSize": "10px",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={
        {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a />
    </div>
    <div
      style={
        {
          "paddingLeft": 20,
        }
      }
    />
  </div>
  <div
    style={
      {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "marginBottom": 1,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        {
          "color": "#888",
          "fontSize": "10px",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={
        {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a />
    </div>
    <div
      style={
        {
          "paddingLeft": 20,
        }
      }
    />
  </div>
</div>
`;
